server/public/plugin/environment.go

```diff
@@ -352,6 +352,71 @@ func (env *Environment) Activate(id string) (manifest *model.Manifest, activated
 	return pluginInfo.Manifest, true, nil
 }
 
+// Reattach allows the server to bind to an existing plugin instance launched elsewhere.
+func (env *Environment) Reattach(manifest *model.Manifest, pluginReattachConfig *model.PluginReattachConfig) (reterr error) {
```

Comment:
Right now, there are non-trivial differences between the two despite the apparent commonality:
* We don't `env.Available` to find the `BundleInfo` but build one tailor-purposed for the reattaching
* We require a server for reattaching, and ignore a webapp with a log when reattaching
* The return value needs are different (and I'm trying to avoid breaking the existing plugin model as much as possible)

Instead of trying to generalize here, I opted to do most of that code sharing via `newSupervisor` with the `With...Config` parameters.

But I'm open to suggestions! What are your thoughts, @crspeller?

Reply:
I ended up factoring out a `checkMinServerVersion` and a `startPluginServer` -- let me know if that helps reduce duplication here.